Rubberduck.Parsing/ParserRuleContextExtensions.cs

```diff
@@ -13,14 +13,15 @@ public static Selection GetSelection(this ParserRuleContext context)
             // if we have an empty module, `Stop` is null
             if (context?.Stop == null) { return Selection.Home; }
 
-            // ANTLR indexes are 0-based, but VBE's are 1-based.
+            // ANTLR indexes for columns are 0-based, but VBE's are 1-based.
+            // ANTLR lines and VBE's lines are both 1-based
             // 1 is the default value that will select all lines. Replace zeroes with ones.
             // See also: https://msdn.microsoft.com/en-us/library/aa443952(v=vs.60).aspx
 
             return new Selection(context.Start.Line == 0 ? 1 : context.Start.Line,
                                  context.Start.Column + 1,
-                                 context.Stop.Line == 0 ? 1 : context.Stop.Line,
-                                 context.Stop.Column + context.Stop.Text.Length + 1);
+                                 context.Stop.Line == 0 ? 1 : context.Stop.EndLine(),
+                                 context.Stop.EndColumn() + 1);
```

Comment:
This *looks* much more "right" than it did. If all tests still pass and navigation doesn't feel off, that'll be it :+1:

Reply:
Yep. It was miles and miles away and IMHO does *not* explain why tests couldn't fail consistently. At least in one case, it was because one of the test had a `SingleOrDefault()` and the excessive selection size caused more than one declaration to qualify and thus throw an exception about having too many matches. 

That said, we have 2 tests; `Context_NotIn_Selection_StartTooLate_OneBased` and `Context_NotIn_Selection_StartTooLate_OneBased` to help guard against one-off errors and also the case when selection is too excessive. With this change, all tests now pass consistently.